shell.nix

```diff
@@ -1,88 +1,72 @@
+    avr ? true,
+    arm ? true,
+    teensy ? true,
+  }:
+    with pkgs; let
+      avrlibc = pkgsCross.avr.libcCross;
 
-  avr_incflags = [
-    "-isystem ${avrlibc}/avr/include"
-    "-B${avrlibc}/avr/lib/avr5"
-    "-L${avrlibc}/avr/lib/avr5"
-    "-B${avrlibc}/avr/lib/avr35"
-    "-L${avrlibc}/avr/lib/avr35"
-    "-B${avrlibc}/avr/lib/avr51"
-    "-L${avrlibc}/avr/lib/avr51"
-  ];
+      avr_incflags = [
+        "-isystem ${avrlibc}/avr/include"
+        "-B${avrlibc}/avr/lib/avr5"
+        "-L${avrlibc}/avr/lib/avr5"
+        "-B${avrlibc}/avr/lib/avr35"
+        "-L${avrlibc}/avr/lib/avr35"
+        "-B${avrlibc}/avr/lib/avr51"
+        "-L${avrlibc}/avr/lib/avr51"
+      ];
 
-  # Builds the python env based on nix/pyproject.toml and
-  # nix/poetry.lock Use the "poetry update --lock", "poetry add
-  # --lock" etc. in the nix folder to adjust the contents of those
-  # files if the requirements*.txt files change
-  pythonEnv = poetry2nix.mkPoetryEnv {
-    projectDir = ./util/nix;
-    overrides = poetry2nix.overrides.withDefaults (self: super: {
-      pillow = super.pillow.overridePythonAttrs(old: {
-        # Use preConfigure from nixpkgs to fix library detection issues and
-        # impurities which can break the build process; this also requires
-        # adding propagatedBuildInputs and buildInputs from the same source.
-        propagatedBuildInputs = (old.buildInputs or []) ++ pkgs.python3.pkgs.pillow.propagatedBuildInputs;
-        buildInputs = (old.buildInputs or []) ++ pkgs.python3.pkgs.pillow.buildInputs;
-        preConfigure = (old.preConfigure or "") + pkgs.python3.pkgs.pillow.preConfigure;
-      });
-      qmk = super.qmk.overridePythonAttrs(old: {
-        # Allow QMK CLI to run "qmk" as a subprocess (the wrapper changes
-        # $PATH and breaks these invocations).
-        dontWrapPythonPrograms = true;
-      });
-    });
-  };
-in
-mkShell {
-  name = "qmk-firmware";
+      # Builds the python env based on nix/pyproject.toml and
+      # nix/poetry.lock Use the "poetry update --lock", "poetry add
+      # --lock" etc. in the nix folder to adjust the contents of those
+      # files if the requirements*.txt files change
+      pythonEnv = poetry2nix.mkPoetryEnv {
+        projectDir = ./util/nix;
+        overrides = poetry2nix.overrides.withDefaults (self: super: {
+          pillow = super.pillow.overridePythonAttrs (old: {
+            # Use preConfigure from nixpkgs to fix library detection issues and
+            # impurities which can break the build process; this also requires
+            # adding propagatedBuildInputs and buildInputs from the same source.
+            propagatedBuildInputs = (old.buildInputs or []) ++ pkgs.python3.pkgs.pillow.propagatedBuildInputs;
+            buildInputs = (old.buildInputs or []) ++ pkgs.python3.pkgs.pillow.buildInputs;
+            preConfigure = (builtins.toString (old.preConfigure or "")) + pkgs.python3.pkgs.pillow.preConfigure;
```

Comment:
Interesting; I did not need to use `builtins.toString` here — why did you add it?

Reply:
Now that you mention it, I'll test again whether it's needed at this current state.

In my first attempts at making a flake shim, that line was complaining about not receiving any string (although the value of `old.preconfigure` **is** a string). I'll check at this "working state" whether it keeps complaining about it.

Reply:
I took away the call to `builtins.toString` as I didn't find it failing anymore.